### ISSUE TITLE
Azure web App deployment fails with "An error occured when request was processed on the remote computer"

### ISSUE THREAD

**Issue by zechariahks** [Microsoft/vsts-tasks#1555]
I am getting below error in Azure Web App deployment task.This is an intermittent issue. I am getting this error when another Azure Web App deployment task in another Release is in progress(Different Web App in the same region and resource group).

2016-04-19T23:35:37.0955595Z Publish-AzureWebsiteProject -Name "PaymentsVoidService-envd" -Package "C:\Agent\7992b274f\EC.NGVL.Payments.PaymentVoidService\Packages\PaymentVoidService.zip" -Slot "production"  -ErrorVariable publishAzureWebsiteError
2016-04-19T23:36:01.7389451Z ##[error](4/19/2016 4:36:01 PM) _**An error occurred when the request was processed on the remote computer.**_

Note: If I queue my two releases in sequencial manner, i.e. trigger second release after first one completes, then i am not seeing this issue. 

Executing Publish-AzureWebsiteProject for the above two web apps in two different powershell sessions working fine. I am observing the issue only in VSO Task.

let me know if I need to provide more details.

**Comment by niadak** [comment]
@zechariahks ,Thanks for reporting the Issue. i have tried to repro the issue in my setup, but not able to do that. I have tried to run two release on parallel and both the release succeeded. Can you please spacify bellow
What is the AzurePowerShell version installed in the Agent machines?
To run two releases in parallel we need two Agent configured in the pool. Are you using two different machine for that?

**Comment by zechariahks** [comment]
Hi,

Thank you for the quick response.Yes, I have two different agents machines configured in the pool. Both the machines are having latest Azure Powershell version 1.3.2. Intermittently, I am getting the below error as well when running two releases in parallel.
##[error]_Web Deploy experienced a connection problem with the server and had to terminate the connection.  Contact your server administrator if the problem persists.  Learn more at: http://go.microsoft.com/fwlink/?LinkId=221672#ERROR_CONNECTION_TERMINATED._

Note: Prior to this web app deployment task, I have another Powershell task which configures my website settings like Appsettings, Custom Domains, Certificates etc., at the end of the configuration. Not sure if this is causing the connection failure.

**Comment by niadak** [comment]
Can you please try the following:
1. Disable the PowerShell task and rung only the WebSite Task. (If the PS task is not a Pre-Req)
2. Try with WebSite under different AzureResourceGroup (if possible)

**Comment by zechariahks** [comment]
Hi,

I have tried the above suggestions, it works fine if I try to deploy the web site in different resource group irrespective of my first runs or not.

Our requirement is to deploy multiple web sites in the same resource group. Is this a limitation for Web Sites?

**Comment by codito** [comment]
@zechariahks I don't think there is any limit on multiple websites in same resource group. I am able to successfully deploy in this scenario.

Can you please try to set `system.debug` variable to `true` in both the releases? With this variable set, you should see additional logging in release log.

**Comment by zechariahks** [comment]
Below is the debug log for Publish Web App step in one release, other release passed.

2016-04-22T09:14:01.4042232Z ##[error]Microsoft.Web.Deployment.DeploymentDetailedException: Web Deploy experienced a connection problem with the server and had to terminate the connection.  Contact your server administrator if the problem persists.  Learn more at: http://go.microsoft.com/fwlink/?LinkId=221672#ERROR_CONNECTION_TERMINATED. ---> System.Xml.XmlException: Root element is missing.
 
2016-04-22T09:14:01.4042232Z ##[error]   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
 
2016-04-22T09:14:01.4092238Z ##[error]   at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
 
2016-04-22T09:14:01.4092238Z ##[error]   at System.Xml.XmlReader.MoveToContent()
 
2016-04-22T09:14:01.4142313Z ##[error]   at Microsoft.Web.Deployment.TraceEventSerializer.Deserialize(Stream responseStream, DeploymentBaseContext baseContext, DeploymentSyncContext syncContext)
 
2016-04-22T09:14:01.4142313Z ##[error]   --- End of inner exception stack trace ---
 
2016-04-22T09:14:01.4142313Z ##[error]   at Microsoft.Web.Deployment.TraceEventSerializer.Deserialize(Stream responseStream, DeploymentBaseContext baseContext, DeploymentSyncContext syncContext)
 
2016-04-22T09:14:01.4192363Z ##[error]   at Microsoft.Web.Deployment.AgentClientProvider.RemoteDestSync(DeploymentObject sourceObject, DeploymentSyncContext syncContext, Nullable`1 syncPass, String syncSessionId)
 
2016-04-22T09:14:01.4192363Z ##[error]   at Microsoft.Web.Deployment.DeploymentObject.SyncToInternal(DeploymentObject destObject, DeploymentSyncOptions syncOptions, PayloadTable payloadTable, ContentRootTable contentRootTable, Nullable`1 syncPassId, String syncSessionId)
 
2016-04-22T09:14:01.4192363Z ##[error]   at Microsoft.Web.Deployment.DeploymentObject.SyncTo(DeploymentProviderOptions providerOptions, DeploymentBaseOptions baseOptions, DeploymentSyncOptions syncOptions)
 
2016-04-22T09:14:01.4342228Z ##[error]   at Microsoft.WindowsAzure.Commands.Utilities.Websites.WebsitesClient.PublishWebProjectFromPackageFile(String websiteName, String slot, String package, String setParametersFile, Hashtable connectionStrings, Boolean skipAppData, Boolean doNotDelete)
 
2016-04-22T09:14:01.4342228Z ##[error]   at Microsoft.WindowsAzure.Commands.Websites.PublishAzureWebsiteProject.ExecuteCmdlet()
 
2016-04-22T09:14:01.4401161Z ##[error]   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()

**Comment by codito** [comment]
Looks like it is same as #471, we've not been able to reproduce this in the past. I will try again.

Are you also trying to deploy to Azure Western Europe region?

There's a blog which mentions network glitch as a possible cause for this: https://blogs.msdn.microsoft.com/benjaminperkins/2015/09/14/error_connection_terminated-msdeploy-web-deploy/

**Comment by zechariahks** [comment]
Hi,

I am deploying to West US region. I think i found the issue. If I use Azure Web App deployment task to publish package to a new web app, it is working fine. It is creating the web App with App Service Plan Default(Free) service plan.

In my case I have already created an App Service Plan with Standard2 Tier with 3 instances. When I use the Azure Web App deployment task on this service plan, it is not working in parallel. Seems to be issue with the instance count I am setting. 

I have removed the customizing the number of instances for the App Service Plan with Standard2 Tier. Azure Web App deployment working in parallel after this change.
